### ISSUE TITLE
incorrect logic for treatment

### ISSUE THREAD

**Issue by gglusman** [NCATSTranslator/Feedback#802]
Testing what may treat sepsis https://ui.test.transltr.io/main/results?l=Sepsis&i=HP:0100806&t=0&r=0&q=d2f631c2-11d6-4ffd-90b1-935f81c7bcdf

Result 2: "methylmercury chloride" (of mad hatter disease fame) is claimed to treat since it increases expression of a gene that causes the phenotype.
![image](https://github.com/NCATSTranslator/Feedback/assets/3977332/068fcb8f-d163-4d62-8e5d-0d246dc6ef93)

**Comment by kaiwenho** [comment]
It might look odd. However, when we refer to the edge (gene-causes-disease), we do not have the whole picture on how the gene contributes to the disease—whether it's due to the gene's presence, its lack of expression, or other mechanisms. For this example/answer, it's possible that the disease is caused by a lack of COTL1 expression. 

mediKanren takes a cautious approach to inference when information is insufficient - the system generates potential answers (chemicals) that might treat a disease based on algorithm and rules, then screens these answers and filters out any chemicals that are known to cause the disease. We want to ensure that we don't overlook these possibilities, especially in the context of rare diseases.

**Comment by gglusman** [comment]
Does it make sense to have a score of 5.0 when the underlying information is insufficient, to the point of yielding the precisely incorrect answer?

**Comment by kaiwenho** [comment]
> Does it make sense to have a score of 5.0 when the underlying information is insufficient, to the point of potentially yielding the precisely incorrect answer?

You’re absolutely right that the scoring is improper in this case. The improved scoring algorithm now incorporates additional factors, including the length of the path (with a preference for look-up > 1-hop inferred > 2-hop inferred), the 'strength' of the predicate (e.g., causation is prioritized over association), answer relatedness (whether the concept from the query or its subclass is treated), and supporting studies (publications). 

I think the new algorithm has fixed the scoring issue for this specific case, currently in [CI](https://ui.ci.transltr.io/results?l=Sepsis&i=HP:0100806&t=0&r=0&q=b386b265-3754-48d6-8f0b-4394480392e2) - PK: b386b265-3754-48d6-8f0b-4394480392e2
<img width="1070" alt="Screenshot 2024-08-15 at 8 43 18 PM" src="https://github.com/user-attachments/assets/241dfbf5-c18e-43d6-8100-a076f0b0ac6a">
